src/main/java/CREResultsRetriever.java

```diff
@@ -63,5 +69,33 @@ private void prepareCREResults() throws IOException {
                 }
             }
         }
+        openFile();
+        addResultsToFile();
+        closeFile();
+    }
+
+    private void openFile() {
+        Date date = new Date();
+        String stringDate = date.toString().replaceAll(":", "-");
+        try {
+            System.out.println("Cr" + stringDate + ".txt");
+            x = new Formatter("dates"+ File.separator+stringDate + "-Currencies Rate.txt");
```

Comment:
I doubt that Formatter is good approach with working with file. There are many ways how to do it in java. I think that better will be something from this list: File object, FileOutputStream, static methods from Files class (Java 8 style).
https://howtodoinjava.com/java/io/how-to-create-a-new-file-in-java/